UIKit/Location Controller/TWTLocationController.m

```diff
@@ -0,0 +1,189 @@
+//
+//  TWTLocationController.m
+//  Toast
+//
+//  Created by Steve Foster on 2/3/15.
+//  Copyright (c) 2015 Two Toasters, LLC. All rights reserved.
+//
+
+#import "TWTLocationController.h"
+
+@import CoreLocation;
+
+static NSTimeInterval const kLocationTimeoutInterval = 60;
+NSString *const kLocationControllerErrorDomain = @"locationcontroller";
+
+
+@interface TWTLocationController () <CLLocationManagerDelegate>
+
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic, strong) CLLocation *location;
+@property (nonatomic, strong) NSMutableArray *requestBlocks;
+
+@end
+
+@implementation TWTLocationController
+
++ (instancetype)sharedInstance
+{
+    static dispatch_once_t onceToken;
+    static TWTLocationController *sharedInstance = nil;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[TWTLocationController alloc] init];
+    });
+    return sharedInstance;
+}
+
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
+        self.locationManager.desiredAccuracy = kCLLocationAccuracyHundredMeters;
```

Comment:
Is this the only way to configure the location controller? This is very limited. What if I want to use the location controller with other accuracies? This locks in how you have to use location manager.